WordPress/Classes/ViewRelated/Blog/SharingAuthorizationWebViewController.swift

```diff
@@ -0,0 +1,503 @@
+}
+
+/*
+#pragma mark - SharingAuthorizationWebViewController
+ */
+
+/**
+ *    @brief    classify actions taken by web API
+ */
+/*
+typedef NS_ENUM(NSInteger, AuthorizeAction) {
+    AuthorizeActionNone,
+    AuthorizeActionUnknown,
+    AuthorizeActionRequest,
+    AuthorizeActionVerify,
+    AuthorizeActionDeny,
+};
+ */
+
+@objc
+class SharingAuthorizationWebViewController: WPWebViewController {
+    /// Classify actions taken by the web API
+    ///
+    private enum AuthorizeAction: Int {
+        case none
+        case unknown
+        case request
+        case verify
+        case deny
+    }
+/*
+static NSString * const SharingAuthorizationLoginURL = @"https://wordpress.com/wp-login.php";
+static NSString * const SharingAuthorizationPrefix = @"https://public-api.wordpress.com/connect/";
+static NSString * const SharingAuthorizationRequest = @"action=request";
+static NSString * const SharingAuthorizationVerify = @"action=verify";
+static NSString * const SharingAuthorizationDeny = @"action=deny";
+ */
+    private static let loginURL = "https://wordpress.com/wp-login.php"
+    private static let authorizationPrefix = "https://public-api.wordpress.com/connect/"
+    private static let requestActionParameter = "action=request"
+    private static let verifyActionParameter = "action=verify"
+    private static let denyActionParameter = "action=deny"
+
+/*
+// Special handling for the inconsistent way that services respond to a user's choice to decline oauth authorization.
+// Tumblr is uncooporative and doesn't respond in a way that clearly indicates failure.
+// Path does not set the action param or call the callback. It forwards to its own URL ending in /decline.
+static NSString * const SharingAuthorizationPathDecline = @"/decline";
+// LinkedIn
+static NSString * const SharingAuthorizationUserRefused = @"oauth_problem=user_refused";
+// Twitter
+static NSString * const SharingAuthorizationDenied = @"denied=";
+// Facebook and Google+
+static NSString * const SharingAuthorizationAccessDenied = @"error=access_denied";
+ */
+    private static let declinePath = "/decline"
+    private static let userRefused = "oauth_problem=user_refused"
+    private static let authorizationDenied = "denied="
+    private static let accessDenied = "error=access_denied"
+
+
+//@interface SharingAuthorizationWebViewController ()
+
+/**
+ *    @brief    verification loading -- dismiss on completion
+ */
+//@property (nonatomic, assign) BOOL loadingVerify;
+    private var loadingVerify: Bool = false
+
+/**
+ *    @brief    publicize service being authorized
+ */
+//@property (nonatomic, strong) PublicizeService *publicizer;
+    private let publicizer: PublicizeService
+
+
+//@property (nonatomic, strong) NSMutableArray *hosts;
+    private var hosts = [String]()
+
+// From old header...
+//@property (nonatomic, weak) id<SharingAuthorizationDelegate> delegate;
+    private let delegate: SharingAuthorizationDelegate
```

Comment:
I guess the delegate should be a `weak` var. Otherwise, we might end with cycle references.

Reply:
Changed!